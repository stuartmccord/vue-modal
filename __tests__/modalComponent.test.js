import { Modal } from '../src';
import Vue from 'vue/dist/vue.js';

describe('Modal', () => {
    Vue.component('modal', Modal);

    beforeEach(() => {
        document.body.innerHTML = `
            <div id="app">
                <modal :show="false"></modal>
            </div>
        `;
    });

    it('can mount', async () => {
        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('uses a custom slot', async () => {
        document.body.innerHTML = `
            <div id="app">
                <modal :show="false">
                    Modal content
                </modal>
            </div>
            `;

        const modal = await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });

    it('it can show', async () => {
        document.body.innerHTML = `
            <div id="app">
                <modal :show="true"></modal>
            </div>
            `;

        const modal = await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });
});

async function createVm() {
    const vm = new Vue({
        el: '#app',
    });

    await Vue.nextTick(() => {});

    return { app: vm, component: vm.$children[0] };
}
