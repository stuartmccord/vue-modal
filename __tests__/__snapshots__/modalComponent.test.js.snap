// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal can mount 1`] = `
"
            <div id=\\"app\\"><div transition=\\"modal\\" class=\\"modal-mask\\" style=\\"display: none;\\"><div class=\\"modal-wrapper\\"><div class=\\"modal-container\\"></div></div></div></div>
        "
`;

exports[`Modal it can show 1`] = `
"
            <div id=\\"app\\"><div transition=\\"modal\\" class=\\"modal-mask\\"><div class=\\"modal-wrapper\\"><div class=\\"modal-container\\"></div></div></div></div>
            "
`;

exports[`Modal uses a custom slot 1`] = `
"
            <div id=\\"app\\"><div transition=\\"modal\\" class=\\"modal-mask\\" style=\\"display: none;\\"><div class=\\"modal-wrapper\\"><div class=\\"modal-container\\">
                    Modal content
                </div></div></div></div>
            "
`;
